Documentation/openshift.md

```diff
@@ -0,0 +1,101 @@
+
+## Rook Privileges
+
+To orchestrate the storage platform, Rook requires the following access in the cluster:
+- Create `hostPath` volumes, for persistence by the Ceph mon and osd pods
+- Run pods in `privileged` mode, for access to `/dev` and `hostPath` volumes
+- Host networking
+
+## Security Context Constraints
+
+Before starting the Rook operator or cluster, create the security context constraints needed by the Rook pods. The following yaml is found in `rook-scc.yaml` under `/cluster/examples/kubernetes`.
+
+```yaml
+kind: SecurityContextConstraints
+apiVersion: v1
+metadata:
+  name: rook
+allowPrivilegedContainer: true
+allowHostNetwork: true
+allowHostDirVolumePlugin: true
+priority: 
+allowedCapabilities: []
+allowHostPorts: false
+allowHostPID: false
+allowHostIPC: false
+readOnlyRootFilesystem: false
+requiredDropCapabilities: []
+defaultAddCapabilities: []
+runAsUser:
+  type: RunAsAny
+seLinuxContext:
+  type: MustRunAs
+fsGroup:
+  type: MustRunAs
+supplementalGroups:
+  type: RunAsAny
+allowedFlexVolumes: 
+  - driver: "rook.io/rook"
+volumes:
+  - configMap
+  - downwardAPI
+  - emptyDir
+  - flexVolume
+  - hostPath
+  - persistentVolumeClaim
+  - projected
+  - secret
+users:
+  # A user needs to be added for each rook service account. 
+  # This assumes running in the default sample "rook" and "rook-system" namespaces.
+  # If other namespaces are configured, the namespaces need to be updated here.
+  - system:serviceaccount:rook-system:rook-agent
+  - system:serviceaccount:rook-system:rook-discover
+  - system:serviceaccount:rook-system:rook-operator
+  - system:serviceaccount:rook:default
+  - system:serviceaccount:rook:rook-ceph-osd
+```
+
+Important to note is that if you plan on running Rook in namespaces other than the defaults of `rook-system` and `rook`, the example scc will need to be modified to accommodate for your namespaces where the Rook pods are running.
+
+To create the scc you will need a privileged account:
+```bash
+oc login -u system:admin
+```
+
+Now create the scc:
+```bash
+oc create -f rook-scc.yaml
+```
+
+## Rook Settings
+
+There are some Rook settings that also need to be adjusted to work in OpenShift.
+
+### Operator Settings
+There are two environment variables that need to be set in the operator spec that will allow Rook to run in OpenShift clusters.
+- `ROOK_HOSTPATH_REQUIRES_PRIVILEGED`: Must be set to `true`. Writing to the hostPath is required for the Ceph mon and osd pods. Given the restricted permissions in OpenShift with SELinux, the pod must be running privileged in order to write to the hostPath volume.
```

Comment:
is this to workaround the anyuid issues https://github.com/rook/rook/issues/1314#issuecomment-355799641?